Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -1961,20 +1961,32 @@ static MTLPixelFormat pixelFormat(WGPUTextureFormat textureFormat)
     }
 }
 
-static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonPrivateDepthStencilTextures)
+static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonPrivateDepthStencilTextures, bool isBackedByIOSurface)
 {
+    // IOSurface-backed texture must be MTLStorageModeManaged, else the driver will assert with
+    // "Texture Descriptor Validation IOSurface textures must use MTLStorageModeManaged"
+    if (isBackedByIOSurface)
+#if PLATFORM(MAC) || PLATFORM(MACCATALYST)
+        return MTLStorageModeManaged;
+#else
+        // MTLStorageModeManaged is not supported on iOS.
+        return MTLStorageModeShared;
+#endif
+
     if (!supportsNonPrivateDepthStencilTextures)
```

Comment:
I know you didn't write this, but I think this logic is wrong. This condition should only be true if the texture we're trying to create is a depth or stencil texture. Can we add in another argument into this function

```
bool isForDepthOrStencil
```

and then change this condition to be

```
if (isForDepthOrStencil && !supportsNonPrivateDepthStencilTextures)
```

Reply:
Maybe for another patch? I don't like that we're doing two things in a commit

Reply:
ok. can you at least add a FIXME then?